prov/tcp/src/tcpx_comm.c

```diff
@@ -51,7 +51,8 @@ int tcpx_send_msg(struct tcpx_xfer_entry *tx_entry)
 		return ofi_sockerr() == EPIPE ? -FI_ENOTCONN : -ofi_sockerr();
 
 	tx_entry->done_len += bytes_sent;
-	if (tx_entry->done_len < ntohll(tx_entry->hdr.base_hdr.size)) {
+	if (tx_entry->done_len <
+	    tx_entry->ep->hdr_sz_ntoh_fn(&tx_entry->hdr.base_hdr)) {
```

Comment:
This argues for having the tx_entry duplicate the size, which you may have had in a previous version. :/ If we add the size back into the tx_entry, we can drop this handler.